--- a/modules/sandbox/thirdparty/libriscv/lib/libriscv/win32/dlfcn.cpp
+++ b/modules/sandbox/thirdparty/libriscv/lib/libriscv/win32/dlfcn.cpp
@@ -640,7 +640,7 @@
     if( optionalHeader->Magic != IMAGE_NT_OPTIONAL_HDR_MAGIC )
         return FALSE;
 
-    if( index < 0 || index >= IMAGE_NUMBEROF_DIRECTORY_ENTRIES || index >= optionalHeader->NumberOfRvaAndSizes )
+    if( index < 0 || index >= IMAGE_NUMBEROF_DIRECTORY_ENTRIES || (DWORD)index >= optionalHeader->NumberOfRvaAndSizes )
         return FALSE;
 
     if( optionalHeader->DataDirectory[index].Size == 0 || optionalHeader->DataDirectory[index].VirtualAddress == 0 )
@@ -904,4 +904,4 @@
 }
 #endif
 
-// Path: lib/libriscv/win32/dlfcn.h
\ No newline at end of file
+// Path: lib/libriscv/win32/dlfcn.h
