--- a/modules/sandbox/thirdparty/libriscv/lib/libriscv/memory_helpers_paging.hpp
+++ b/modules/sandbox/thirdparty/libriscv/lib/libriscv/memory_helpers_paging.hpp
@@ -215,7 +215,7 @@
 
 #ifdef RISCV_VIRTUAL_PAGING
 	// Fallback: Try gathering a single buffer, which throws OUT_OF_MEMORY if it fails
-	std::array<vBuffer, 1> buffers;
+	std::array<vBuffer, 1> buffers{};
 	gather_buffers_from_range(1, buffers.data(), addr, len);
 	return {(const char *)buffers[0].ptr, buffers[0].len};
 #else
@@ -241,7 +241,7 @@
 
 #ifdef RISCV_VIRTUAL_PAGING
 	// Fallback: Try gathering a single buffer, which throws OUT_OF_MEMORY if it fails
-	std::array<vBuffer, 1> buffers;
+	std::array<vBuffer, 1> buffers{};
 	const_cast<Memory<W>*> (this)->gather_writable_buffers_from_range(1, buffers.data(), addr, len);
 	return {(char *)buffers[0].ptr, buffers[0].len};
 #else
