--- a/modules/sandbox/thirdparty/libriscv/lib/libriscv/native_heap.hpp
+++ b/modules/sandbox/thirdparty/libriscv/lib/libriscv/native_heap.hpp
@@ -29,9 +29,9 @@
 	PointerType data = 0;
 
 	ArenaChunk* find_used(PointerType ptr) const;
-	ArenaChunk* find_free(size_t size);
+	ArenaChunk* find_free(size_t req_size);
 	void merge_next(Arena&);
-	void split_next(Arena&, size_t size);
+	void split_next(Arena&, size_t chunk_size);
 	void subsume_next(Arena&, size_t extra);
 };
 
@@ -187,11 +187,11 @@
 	return nullptr;
 }
 // find free chunk that has at least given size
-inline ArenaChunk* ArenaChunk::find_free(size_t size)
+inline ArenaChunk* ArenaChunk::find_free(size_t req_size)
 {
     ArenaChunk* ch = this;
 	while (ch != nullptr) {
-		if (ch->free && ch->size >= size)
+		if (ch->free && ch->size >= req_size)
 			return ch;
 		ch = ch->next;
 	}
@@ -233,17 +233,17 @@
 	}
 }
 
-inline void ArenaChunk::split_next(Arena& arena, size_t size)
+inline void ArenaChunk::split_next(Arena& arena, size_t chunk_size)
 {
 	// Only split if the new chunk would not be empty
-	if (this->size > size)
+	if (this->size > chunk_size)
 	{
 		ArenaChunk* newch = arena.new_chunk(
 			this->next,
 			this,
-			this->size - size,
+			this->size - chunk_size,
 			true, // free
-			this->data + (PointerType) size
+			this->data + (PointerType) chunk_size
 		);
 		if (this->next) {
 			this->next->prev = newch;
@@ -260,7 +260,7 @@
 			this->next->prev = nullptr;
 		}
 	}
-	this->size = size;
+	this->size = chunk_size;
 }
 
 template <typename... Args>
