--- a/modules/sandbox/thirdparty/libriscv/lib/libriscv/win32/dlfcn.h
+++ b/modules/sandbox/thirdparty/libriscv/lib/libriscv/win32/dlfcn.h
@@ -40,12 +40,9 @@
 
 /* Enable weak symbols to account for Godots own dlsym and dlerror
  * implementations. This is needed to avoid linking errors when Godot
- * is built with dlfcn-win32. Unfortunately, this doesn't work when
- * building with -fPIC, so we need to account for that too.
+ * is built with dlfcn-win32.
  */
-#if (defined(__MINGW32__) || defined(__MINGW64__)) && \
-	(defined(__GNUC__) || defined(__clang__)) && \
-	!defined(__PIC__) && !defined(__pic__)
+#ifdef GODOT_MODULE
 #define  DLFCN_WEAK __attribute__((weak))
 #else
 #define  DLFCN_WEAK
@@ -106,4 +103,4 @@
 
 #endif /* DLFCN_H */
 
-// Path: lib/libriscv/win32/dlfcn.c
\ No newline at end of file
+// Path: lib/libriscv/win32/dlfcn.c
