--- a/modules/sandbox/thirdparty/libriscv/lib/libriscv/util/threadpool.h
+++ b/modules/sandbox/thirdparty/libriscv/lib/libriscv/util/threadpool.h
@@ -256,8 +256,8 @@
                 bool notify;
 
                 {
-                    std::unique_lock<std::mutex> lock(this->queue_mutex);
-                    this->condition_consumers.wait(lock,
+                    std::unique_lock<std::mutex> worker_func_lock(this->queue_mutex);
+                    this->condition_consumers.wait(worker_func_lock,
                         [this, worker_number]{
                             return this->stop || !this->tasks.empty()
                                 || pool_size < worker_number + 1; });
@@ -292,7 +292,7 @@
 
                 if (notify)
                 {
-                    std::unique_lock<std::mutex> lock(this->queue_mutex);
+                    std::unique_lock<std::mutex> notify_lock(this->queue_mutex);
                     condition_producers.notify_all();
                 }
 
